### PR TITLE
Add test for internal/config/mysql

### PR DESCRIPTION
**Pull request by hlts2**
### Description:

I added test for `internal/config/mysql_test`. and godoc comment.

**grammer check passed**

### Environment:

- Go Version: 1.16
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.3

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [ ] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: internal/config/mysql.go [hunk 1]
```diff
@@ -73,6 +73,8 @@ func (m *MySQL) Bind() *MySQL {
 	return m
 }
 
+// Opts creates and returns the slice with the functional options for the internal mysql package.
+// In addition, Opts returns the error when the any errors are occurred.
```
Comment (hlts2) [inline internal/config/mysql.go:77]:
```suggestion
// Opts creates and returns the slice with the functional options for the internal mysql package.
// In addition, Opts returns the error when any errors occur.
```

File: internal/config/mysql_test.go [hunk 2]
```diff
@@ -286,71 +291,174 @@ func TestMySQL_Opts(t *testing.T) {
 		if !errors.Is(err, w.err) {
 			return errors.Errorf("got_error: \"%#v\",\n\t\t\t\twant: \"%#v\"", err, w.err)
 		}
-		if !reflect.DeepEqual(got, w.want) {
-			return errors.Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", got, w.want)
+		if !reflect.DeepEqual(len(got), len(w.want)) {
+			return errors.Errorf("length got: \"%#v\",\n\t\t\t\twant: \"%#v\"", len(got), len(w.want))
```
Comment (vankichi) [inline internal/config/mysql_test.go:295]:
```suggestion
			return errors.Errorf("length got: \"%#v\",\n\t\t\t\tlenght want: \"%#v\"", len(got), len(w.want))
```
if you'd like to output of length, you have to change like above.

File: internal/config/mysql_test.go [hunk 3]
```diff
@@ -286,71 +291,174 @@ func TestMySQL_Opts(t *testing.T) {
...
+					CA:      testdata.GetTestdataPath("tls/dummyCa.pem"),
+				},
+				Net: new(Net),
+			},
+			want: want{
+				want: make([]mysql.Option, 17),
+			},
+		},
+		{
+			name: "return 16 option and nil when all parameters are set but the network type is invalid",
```
Comment (vankichi) [inline internal/config/mysql_test.go:330]:
```suggestion
			name: "return 17 option and nil when all parameters are set but the network type is invalid",
```

File: internal/config/mysql_test.go [hunk 4]
```diff
@@ -286,71 +291,174 @@ func TestMySQL_Opts(t *testing.T) {
...
+					CA:      testdata.GetTestdataPath("tls/dummyCa.pem"),
+				},
+				Net: new(Net),
+			},
+			want: want{
+				want: make([]mysql.Option, 17),
+			},
+		},
+		{
+			name: "return 16 option and nil when all parameters are set but the network type is empty",
```
Comment (vankichi) [inline internal/config/mysql_test.go:359]:
```suggestion
			name: "return 17 option and nil when all parameters are set but the network type is empty",
```

File: internal/config/mysql_test.go [hunk 5]
```diff
@@ -286,71 +291,174 @@ func TestMySQL_Opts(t *testing.T) {
...
-		           TLS: TLS{},
-		           Net: Net{},
-		           },
-		           want: want{},
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+		{
+			name: "return 16 option and nil when all parameters are set",
```
Comment (vankichi) [inline internal/config/mysql_test.go:301]:
```suggestion
			name: "return 17 option and nil when all parameters are set",
```

File: internal/config/mysql_test.go [hunk 6]
```diff
@@ -286,71 +291,174 @@ func TestMySQL_Opts(t *testing.T) {
 		if !errors.Is(err, w.err) {
 			return errors.Errorf("got_error: \"%#v\",\n\t\t\t\twant: \"%#v\"", err, w.err)
 		}
-		if !reflect.DeepEqual(got, w.want) {
-			return errors.Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", got, w.want)
+		if !reflect.DeepEqual(len(got), len(w.want)) {
+			return errors.Errorf("length got: \"%#v\",\n\t\t\t\tlenght want: \"%#v\"", len(got), len(w.want))
```
[the remark this answers is not in the document]
Comment (vankichi) [reply internal/config/mysql_test.go:295]: please fix this

[the remark this answers is not in the document]
Comment (hlts2) [reply internal/config/mysql_test.go:295]: thank you.

[the remark this answers is not in the document]
Comment (hlts2) [reply internal/config/mysql_test.go:295]:
```suggestion
			return errors.Errorf("length got: \"%#v\",\n\t\t\t\tlength want: \"%#v\"", len(got), len(w.want))
```

**Review by vankichi — APPROVED** [review]

**Review by kevindiu — DISMISSED** [review]
LGTM

**Pull request merged**